packages/babel-helpers/src/helpers/AsyncGenerator.ts

```diff
@@ -0,0 +1,155 @@
+/* @minVersion 7.0.0-beta.0 */
+
+import OverloadYield from "./OverloadYield.ts";
+
+type AsyncIteratorMethod = "next" | "throw" | "return";
+
+declare class AsyncGeneratorImpl<T = unknown, TReturn = any, TNext = unknown>
+  implements AsyncGenerator<T, TReturn, TNext>
```

Comment:
We usually try to keep the function name the same as the helper name, so that it looks "as native as possible", at least when using `@babel/runtime`. e.g.
```js
(async function* f() {}).name === "AsyncGenerator"
```

(it's not something we guarantee, it's more of a nice-to-have).

Could you this instead?
```ts
declare class AsyncGenerator<T = unknown, TReturn = any, TNext = unknown>
  implements globalThis.AsyncGenerator<T, TReturn, TNext>
```